### ISSUE TITLE
[C++] TopicImpl::name() creates string copies every time

### ISSUE THREAD

**Issue by iolazarc** [edenhill/librdkafka#2682]
Read the FAQ first: https://github.com/edenhill/librdkafka/wiki/FAQ

Description
===========
Every time TopicImpl::name() is called, a new std::string is created via copy form C string. That will waste CPU and create extra allocations/deallocations if someone decides to do things like log the topic name from a Topic instance. Topic::name() should return a std::string&, like TopicPartition::topic() does. TopicImpl::name() should create the string lazily and store it as a member for reuse.

How to reproduce
================
In code.

**IMPORTANT**: Always try to reproduce the issue on the latest released version (see https://github.com/edenhill/librdkafka/releases), if it can't be reproduced on the latest version the issue has been fixed.

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed.

Please provide the following information:

 - [x] librdkafka version (release number or git tag): 1.3
 - [ ] Apache Kafka version: n/a
 - [ ] librdkafka client configuration: n/a
 - [ ] Operating system: n/a
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by edenhill** [comment]
Would you like to submit a PR where a new method is added to return a string reference?
either `std::string &name_ref(), or perhaps `name(std::string &name)`?

**Comment by iolazarc** [comment]
Yes, although I don't know exactly when I'll get around to this.

**Comment by MikeWezovski** [comment]
!(O_o)!
